product_contract/wizards/product_contract_configurator.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2024 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class ProductContractConfigurator(models.TransientModel):
+    _name = "product.contract.configurator"
+    _description = "Product Contract Configurator Wizard"
+
+    product_id = fields.Many2one("product.product")
+    partner_id = fields.Many2one("res.partner")
+    company_id = fields.Many2one("res.company")
+    product_uom_qty = fields.Float("Quantity")
+    contract_id = fields.Many2one(comodel_name="contract.contract", string="Contract")
+    contract_template_id = fields.Many2one(
+        comodel_name="contract.template",
+        string="Contract Template",
+        compute="_compute_contract_template_id",
+    )
+    recurring_rule_type = fields.Selection(
```

Comment:
Now that we have configured all the start date thing in the product, it doesn't seem OK to also have again this configuration here, don't you think?

Reply:
I think we should keep it in case the product is set to manual mode, but the unnecessary fields should be hidden in the view.

Reply:
@sbejaoui which was the reason to define a recurrency in the product and ask for that recurrency again in the sales order line? Can we strip that part for simplifying code and UI?